anxcloud/testutils/recorder/vm_recorder.go

```diff
@@ -0,0 +1,79 @@
+package recorder
+
+import (
+	"context"
+	"errors"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/client"
+	"go.anx.io/go-anxcloud/pkg/vsphere/provisioning/vm"
+	"go.anx.io/go-anxcloud/pkg/vsphere/search"
+	"net/http"
+)
+
+type vmCleanUpFunc func(ctx context.Context) error
+type VMRecoder struct {
+	handlers []vmCleanUpFunc
+	client   api.API
+}
+
+func (v VMRecoder) Cleanup(ctx context.Context) []error {
+	var cleanUpErrors []error
+	for _, handler := range v.handlers {
+		err := handler(ctx)
+		if err != nil {
+			cleanUpErrors = append(cleanUpErrors, err)
+		}
+	}
+	return nil
+}
+
+func (v *VMRecoder) RecordVMByName(name string) {
+	v.handlers = append(v.handlers, v.createCleanupHandlerByName(name))
+}
+
+func (v *VMRecoder) RecordVMByID(id string) {
+	v.handlers = append(v.handlers, v.createCleanupHandlerByName(id))
```

Comment:
ByName called in ByID

Reply:
Was fixed once then broke during rebasing again :/